### PR TITLE
Enhancement editor loading speed

### LINKED ISSUE #46491: [Enhancement]: Optimize Large Contentful Paint Elements
- `Summary` field: Identified as one of the largest contentful paint elements, optimizing this field can significantly enhance rendering speed.

summary-Screenshot 2024-04-09 at 11 53 57 AM

Review and optimize the underlying code for the Summary field.
Implement more efficient CSS and JavaScript, ensuring that heavy computations or styles while block rendering are minimized or eliminated.

### PR DESCRIPTION
**Pull request by mdperez86**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Closes #46491

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure the new product editor is enabled under the WooCommerce > Settings > Advance > Features
2. Then try to edit a product and see that everything works as expected.
3. Check if the tab behavior is correct.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

**Comment by octaedro** [comment]
This is looking great @mdperez86! I'm testing this PR, but after enabling GB, the product editor stops working.

**Comment by mdperez86** [comment]
> This is looking great @mdperez86! I'm testing this PR, but after enabling GB, the product editor stops working.

It looks like this has nothing to do with this PR but I have a fix for it [here](https://github.com/woocommerce/woocommerce/pull/47678) anyway.

**Review by octaedro — APPROVED** [review]
Great job @mdperez86! This is testing well on my end and the code looks good!

File: packages/js/product-editor/src/hooks/use-product-url.ts [hunk 1]
```diff
@@ -2,23 +2,27 @@
...
 		productType,
 		'permalink'
 	);
+
 	const getProductURL = useCallback(
 		( isPreview: boolean ) => {
-			const productURL = new URL( permalink ) as URL | undefined;
+			if ( ! permalink ) return undefined;
+
+			const productURL = new URL( permalink );
```
Comment (octaedro) [inline packages/js/product-editor/src/hooks/use-product-url.ts:18]: ❤️

**Pull request merged**